File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.java
class Solution {
    StringBuilder str;
    public String processStr(String s) {
        this.str = new StringBuilder(s);
        StringBuilder temp = new StringBuilder("");
        return this.func(temp,0).toString();

    }

    private StringBuilder func(StringBuilder s, int i){
        if(i == this.str.length()) return s;
        char c = this.str.charAt(i); 
        if(c == '*'){
            if(s.length() != 0){
                s.deleteCharAt(s.length()-1);
            }

        }else if(c == '#'){
            s.append(s);
        }else if(c == '%'){
            s.reverse();
        }else{
            s.append(c);
        }
        return this.func(s,i+1);
        

    }
}

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string str;
    string processStr(string s) {
        this->str = s;
        return this->func("",0);

    }

    string func(string temp, int i){
        if(i == this->str.size()) return temp;
        auto c = this->str[i];
        if(c == '*'){
            if(temp != ""){
                temp.pop_back();
            }
        }else if(c == '#'){
            temp += temp;
        }else if(c == '%'){
            reverse(temp.begin(), temp.end());
        }else{
            temp += c;
        }
        return this->func(temp,i+1);
    }
};